drivers/at_cmd/at_cmd.c

```diff
@@ -284,27 +301,34 @@ int at_cmd_write(const char *const cmd,
 		 size_t buf_len,
 		 enum at_cmd_state *state)
 {
-	int return_code;
+	int return_code = k_sem_take(&cmd_pending, K_FOREVER);
```

Comment:
`k_sem_take` will not return an error if it's called with `K_FOREVER` timeout. It can only return an error code, in case it times-out (-EAGAIN), or when it was called with `K_NO_WAIT` (-EBUSY). Hence, I'd remove error checks to simplify the logic.

Reply:
Ok, removed the code in this function, at_cmd_write_with_callback() and at_cmd_set_notification_handler